systems/WikibaseRepo/04-Solution_Strategy.md

```diff
@@ -1 +1,21 @@
 # Solution Strategy
+
+## Developing Wikibase Repo as a MediaWiki extension
+
+One of the main requirements of Wikibase is to make data available to MediaWiki applications. This makes it a natural choice to deliver the client (= data consumer) functionality as a MediaWiki extension. (see [this section in WikibaseClient](../WikibaseClient/04-Solution_Strategy.md#developing-wikibase-client-as-a-mediawiki-extension)) The same is not necessarily the case for the Wikibase Repository, the subsystem that provides and maintains the structured data. WikibaseRepo could have been developed independently of MediaWiki, yet it was decided to use MediaWiki like a web framework and add the functionality to store and maintain structured data through the WikibaseRepo MediaWiki extension.
+
+This came with the following benefits:
+
+- better compatibility with the rest of the Wikimedia ecosystem
+- achieving a familiar Wikimedia user experience out of the box
+- reduction of initial boilerplate code and basic features
+- the original development team had extensive MediaWiki experience
+- ability to easily have one application that is both a client and a repo
+
+It also had negative consequences:
+
+- hacks to make MediaWiki do things it was not meant to do
+  - storing structured data in the same way Wikitext is stored
+  - showing different content dependent on user settings (language)
```

Comment:
I'm not sure if this is a negative consequence? This is desired?

Reply:
negative or not: is it a "hack to make MediaWiki do things it was not originally designed to do"?

Reply:
Well, I disagree that it is a hack really.

You could say this about any framework?
No framework is designed  to magically show different output from a single block of content dependent on some user language? Whatever framework this happens in some code (not a hack) has to exist to do so.

MediaWiki allows you to render / parse a page with any number of options applied (by design).
One of these is language.
Thus different content appears depending on user settings.

This is the same mechanism that is used for rendering or not rendering the new termbox.

IMO this is not a hack, it is just a feature that we are using?
No different to having to write our own conditional parsing / rendering logic if we were using some other framework.

`request-> check user language -> render page for user using language as an option`